app/models/services/service_instance.rb

```diff
@@ -217,6 +217,10 @@ def has_keys?
       !service_keys.empty?
     end
 
+    def has_routes?
+      !routes.empty?
```

Comment:
```suggestion
      routes.present?
```

Alternative way provided by ActiveSupport, FYI.